src/core/system.h

```diff
@@ -36,17 +36,23 @@ class System {
      * This constructor is not (and should not be) directly called by application code.
      *
      * @param parent `MavsdkImpl` dependency.
-     * @param system_id System id.
-     * @param comp_id Component id.
-     * @param connected If true then the system doesn't wait for heartbeat to go into connected
-     * state
      */
-    explicit System(MavsdkImpl& parent, uint8_t system_id, uint8_t comp_id, bool connected);
+    explicit System(MavsdkImpl& parent);
     /**
      * @brief Destructor.
      */
     ~System();
 
+    /**
+     * @brief Initialize the system.
+     *
+     * @param system_id System id.
+     * @param component_id Component id.
+     * @param connected If true then the system doesn't wait for heartbeat to go into connected
+     * state
+     */
+    void init(uint8_t system_id, uint8_t component_id, bool connected) const;
```

Comment:
I think this should be private, right? Because users of the API should not be able to call this, right?

Reply:
It used to be in the constructor, so it was exposed :thinking:. I don't think it can be private. The alternative is to use a factory, so that it's clear for the user how to instantiate it...

Reply:
Ok you're right. But please add a comment that this is to be used internally only.